### PR TITLE
feat(mt): Baidu machine translation support

### PR DESCRIPTION
**Pull request by wfjsw**

### CODE REVIEW

File: backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduApiService.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    )
+
+    return response.body?.transResult?.first()?.dst
+      ?: throw RuntimeException(response.toString())
+  }
+
+  private fun getSalt(): String {
+    val charPool: List<Char> = ('a'..'z') + ('A'..'Z') + ('0'..'9')
+    return (1..16)
+      .map { Random.nextInt(0, charPool.size).let { charPool[it] } }
```
Comment (JanCizmar) [inline backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduApiService.kt:56]: Kotlin has a method for this `list.random()`. 😎

File: backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    "ro-ro",
+    "hu",
+    "hu-hu",
+  )
+
+  // https://fanyi-api.baidu.com/doc/21
+  // extracted commonly used only
+  val languageTagMap = mapOf(
+ // "yue"
+    "yue-hans" to "yue",
```
Comment (JanCizmar) [inline backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt:117]: [tag]-[region] automatically fallbacks to [tag]. 

Check: io.tolgee.component.machineTranslation.LanguageTagConvertor#findSuitableTag

**Review by JanCizmar — CHANGES_REQUESTED** [review]

**Comment by JanCizmar** [comment]
Thanks a lot!

File: backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt [hunk 3]
```diff
@@ -30,167 +30,98 @@ class BaiduTranslationProvider(
 
   override val supportedLanguages = arrayOf(
     "yue",
-    "yue-hans",
```
Comment (JanCizmar) [inline backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt]: No, no... Don't remove it from supported languages. We don't want fallback when not necessary.

File: backend/app/src/test/kotlin/io/tolgee/api/v2/controllers/TranslationSuggestionControllerTest.kt [hunk 4]
```diff
@@ -233,7 +246,7 @@ class TranslationSuggestionControllerTest : ProjectAuthControllerTest("/v2/proje
 
   @Test
   @ProjectJWTAuthTestMethod
-  fun `it suggests using just enabled services (Google, AWS, DeepL, Azure)`() {
+  fun `it suggests using just enabled services (Google, AWS, DeepL, Azure, Baidu)`() {
     mockDefaultMtBucketSize(4000)
```
Comment (wfjsw) [inline backend/app/src/test/kotlin/io/tolgee/api/v2/controllers/TranslationSuggestionControllerTest.kt:250]: Why this failed

**Comment by JanCizmar** [comment]
For the ktlint: there is a gradle task `ktlintFormat`, which you can run to automatically format the code.

File: backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    "ko" to "kor",
+    // "th"
+    // "pt"
+    // "el"
+    // "bul"
+    "bg" to "bul",
+    // "fin"
+    "fi" to "fin",
+    // "slo"
+    "sk" to "slo",
```
Comment (Martin005) [inline backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt:127]: "sk" is Slovak, but "slo" is Slovenian, this mapping is not correct. Tag for Slovenian is "sl".

File: backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    // "zh"
+    "zh-hans" to "zh",
+    // "wyw"
+    // "fra"
+    "fr" to "fra",
+    // "ara"
+    "ar" to "ara",
+    // "de"
+    // "nl"
+    "nl" to "nl",
```
Comment (Martin005) [inline backend/data/src/main/kotlin/io/tolgee/component/machineTranslation/providers/BaiduTranslationProvider.kt:140]: This mapping is superfluous

**Review by JanCizmar — APPROVED** [review]

**Pull request merged**